src/Versioned.php

```diff
@@ -2479,24 +2521,29 @@ public static function prepopulate_versionnumber_cache($class, $stage, $idList =
      * @template T of DataObject
      * @param class-string<T> $class The name of the class.
      * @param string $stage The name of the stage.
-     * @param string $filter A filter to be inserted into the WHERE clause.
-     * @param string $sort A sort expression to be inserted into the ORDER BY clause.
-     * @param int $limit A limit on the number of records returned from the database.
      * @param string $containerClass The container class for the result set (default is DataList)
      *
      * @return DataList<T> A modified DataList designated to the specified stage
      */
     public static function get_by_stage(
-        $class,
-        $stage,
-        $filter = '',
-        $sort = '',
-        $limit = null,
-        $containerClass = DataList::class
-    ) {
-        ReadingMode::validateStage($stage);
+        string $class,
+        string $stage,
+        string|array $filter = '',
+        string|array|null $sort = '',
+        string|array|null $limit = null,
+        string $containerClass = DataList::class
+    ): DataList {
         $result = DataObject::get($class, $filter, $sort, $limit, $containerClass);
-        return $result->setDataQueryParam([
+        return static::updateListForStage($result, $stage);
+    }
+
+    /**
+     * Update an existing DataList to only include records for a given stage.
+     */
+    public static function updateListForStage(DataList $list, string $stage): DataList
```

Comment:
```suggestion
    public static function updateDataListToFilterByStage(DataList $list, string $stage): DataList
```
And update references to it

Reply:
Done - though `DataList` feels overly verbose, so kept as `updateList...`.

Reply:
Actually, not done. This isn't accurate. `stage_unique` (which doesn't have a corresponding method) would have this name, but that's not how this one works. See the unit tests.

Reply:
It's not clear what this does

Depending on what it does, use one of the following:
- updateListToOnlyIncludeStage (i.e. filters existing records)
- upddateListToAlsoIncludeStage (i.e. adds new records)

Reply:
Done